util/stmtsummary/statement_summary.go

```diff
@@ -982,6 +985,19 @@ func (ssElement *stmtSummaryByDigestElement) toDatum(ssbd *stmtSummaryByDigest)
 	)
 }
 
+func (ssElement *stmtSummaryByDigestElement) decodePlan() string {
+	plan, err := plancodec.DecodePlan(ssElement.samplePlan)
+	if err != nil {
+		if ssElement.samplePlan == discardPlanCauseTooLong {
+			plan = discardPlanCauseTooLong
+		} else {
+			logutil.BgLogger().Error("decode plan in statement summary failed", zap.String("plan", ssElement.samplePlan), zap.String("query", ssElement.sampleSQL), zap.Error(err))
```

Comment:
This will leak sensitive information even when log redaction is on?

Reply:
No, If the log redaction is on, `ssElement.sampleSQL` was already been redact, and `ssElement.samplePlan` since can't been decoded, so it will not leak sensitive information.